app/models/bulk_update/placement_row.rb

```diff
@@ -39,4 +39,12 @@ class BulkUpdate::PlacementRow < ApplicationRecord
     imported: 2,
     failed: 3,
   }
+
+  def can_be_imported?
+    pending? || failed?
+  end
+
+  def row_error_messages
+    row_errors.map(&:message).join("\n")
```

Comment:
i don't feel super-strongly about it but perhaps there's a case for doing things like `.join("\n")` in the bit of the code that deals with handling CSV

Reply:
at the moment it's only the tests that make use of this. agreed though, can move it somewhere more appropriate once we handle unhappy paths 👍